File: 1605-find-valid-matrix-given-row-and-column-sums/1605-find-valid-matrix-given-row-and-column-sums.py
class Solution:
    def restoreMatrix(self, rowSum: List[int], colSum: List[int]) -> List[List[int]]:
        rows, cols = len(rowSum), len(colSum)
        grid = [([0] * cols) for _ in range(rows)]
        for r in range(rows):
            for c in range(cols):
                min_val = min(rowSum[r], colSum[c])
                rowSum[r] -= min_val
                colSum[c] -= min_val
                grid[r][c] = min_val
        
        return grid
        

File: 1605-find-valid-matrix-given-row-and-column-sums/1605-find-valid-matrix-given-row-and-column-sums.go
func restoreMatrix(rowSum []int, colSum []int) [][]int {
    var ret = [][]int{}
    var rows, cols = len(rowSum), len(colSum)
    for i := range rows {
        ret = append(ret, make([]int, cols))
        ret[i][0] = rowSum[i]
    }
    
    for c := range cols - 1{
        for r := range rows {
            if ret[r][c] <= colSum[c] {
                colSum[c] -= ret[r][c]
            } else {
                ret[r][c + 1] = (ret[r][c] - colSum[c])
                ret[r][c] = colSum[c]
                colSum[c] = 0
            }
        }
    }
    
    return ret
}